Cleanly terminating Pilot and Ray

drelu opened the pull request:
- Cleanly terminating Ray and Pilot
- Some work on CC mini app

drelu committed:
pending commits

drelu committed:
Merge branch 'main' of github.com:radical-cybertools/quantum-mini-apps

drelu committed:
Improved pilot quantum executor to shutdown ray correctly

pradeepmantha commented on src/mini_apps/quantum_simulation/circuit_cutting/cc_local_cpu.py:
```diff
@@ -12,6 +18,34 @@
+            # Process may have terminated or we don't have permissions
+            pass
+
+def stop_ray():
```
Pilot-Quantum when starts do ray - stop.. 


https://github.com/radical-cybertools/pilot-quantum/blob/ray-v2/pilot/plugins/ray_v2/cluster.py#L36

pradeepmantha replied:
i am not sure why we need this here again?

pradeepmantha commented on src/engine/cluster/pilot_quantum_executor.py:
```diff
@@ -25,9 +16,24 @@ def __init__(self, cluster_config=None):
+        if self.type == "dask":
+            self.client.close()
+        elif self.type == "ray":
+            ray.shutdown()
```
the pilot.cancel which abstracts dask/ray engine should clean it. this seems like leaking pilot-abstraction into mini-apps.

pradeepmantha replied:
The pilot-cancel already does this - https://github.com/radical-cybertools/pilot-quantum/blob/ray-v2/pilot/plugins/ray_v2/cluster.py#L136

drelu committed:
fixes to log handler

drelu committed:
Merge remote-tracking branch 'origin/main' into develop

drelu committed:
formatting with black formatter

drelu committed:
CC Mini-App improve experimental data written to csv

drelu committed:
Enhance  circuit cutting with additional parameters  and update circuit sizes for testing

drelu commented:
Hi @pradeepmantha, understand that these things should go into Pilot-Quantum. Can you nevertheless merge the branch as it contains substantial further improvements to the circuit cutting mini-app. I suggest to remove the "hacks" after further testing of all mini apps and the paper deadline.

drelu committed:
Update .gitignore to exclude .npy and .csv files; modify circuit sizes and subcircuit size calculation in cc_local_cpu.py; enhance error handling and metrics writing in circuit_cutting_motif.py.

drelu committed:
Refactor quantum simulation circuit cutting:
- Introduced scenario labeling for better tracking of simulation runs.
- Updated backend options for full circuit and circuit cutting simulations.
- Modified circuit sizes for benchmarking and enhanced error handling.
- Removed obsolete config and logging setup files to streamline the codebase.
